collect_app/src/main/java/org/odk/collect/android/activities/viewmodels/CurrentProjectViewModel.kt

```diff
@@ -23,19 +26,36 @@ class CurrentProjectViewModel(
         currentProjectProvider.setCurrentProject(project.uuid)
         Analytics.log(AnalyticsEvents.SWITCH_PROJECT)
         analyticsInitializer.initialize()
-        refresh()
+        updateCurrentProjectIfNeeded()
     }
 
     fun refresh() {
+        updateCurrentProjectIfNeeded()
+
+        if (!File(storagePathProvider.getProjectRootDirPath()).exists()) {
+            Analytics.log(AnalyticsEvents.RECREATE_PROJECT_DIR)
```

Comment:
I thought about a call that works out exactly what has been deleted, but I realized that it wouldn't work properly due to our application initialization recreating the `projects` dir and obscuring the cases where the user has deleted the external files dir or cleared it out.